### ISSUE TITLE
Audit and reduce complexity of accessibility node tree

### ISSUE THREAD

**Issue by BenHenning** [RaspberryPiFoundation/blockly#9304]
### Check for duplicates

- [x] I have searched for similar issues before opening a new one.

### Problem

Ongoing screen reader experimentation work (https://github.com/google/blockly-keyboard-experimentation/discussions/673) has resulted in a lot of changes to the accessibility node tree. This means that there are many extra nodes, incorrectly placed nodes, or incorrectly represented nodes in the tree.

### Request

The accessibility node tree should be configured such that it contains the exact structure desired for screen reader representation. Much of this is covered in other ARIA work so this issue covers ensuring that, once the core functionality is completed, that the structure of the tree is exactly as desired (that is, all nodes are intentionally placed and correct with no extra, invalid, or missing nodes).

Part of this should ensure hierarchy: some nodes may require adding `aria-owns` attributes in places to ensure that ownership is correctly represented.

Part of this should also including auditing all existing ARIA attributes and ensuring each one is correctly and intentionally set (since some existing attributes in core Blockly may require changes).

### Alternatives considered

The only alternative is allowing for excess nodes to be present. This isn't desirable since it could result in some screen readers incorrectly representing the structure of the page, potentially confusing users.

**Comment by rachel-fenichel** [comment]
@BenHenning I've moved this to an in-progress P1 because you already have a PR open to do some of it, and i don't want to let that PR get too stale.

**Comment by BenHenning** [comment]
Fixed in #9449 (well enough anyway).
